### DISCUSSION TITLE
Teleport to other Wolds and send other

### DISCUSSION THREAD

**Discussion by TheBeast2155** [Zrips/CMI#3079] [Feature Requests]
**Description of feature/improvement request:**

---
### Detailed explanation of what it should do, when and in what situation: ###

I Will like to have /tpw so we cant teleport to a other wold and a /tpw %player% %wold% so we also Can send other players

**Comment by mrfloris** [comment]
```
/cmi rt
/cmi tpallworld
/cmi tppos
/cmi tpopos
/cmi world
and /cmi tp / tpa
```

These all support world teleporting?

You can make a custom alias to use /tpw if you want, with a command like:
`asConsole! cmi tppos -p:[playerName] 100 70 100 %world% -rng`

```
tppos (-p:playerName) [x] [y] [z] (world) (pitch) (yaw) (-rng:[range])
Teleports to location

-p: will define player you want to teleport. This will require you to have cmi.command.tppos.others permission node

-rng: will teleport at random location from center point by given radius

x, y and z can be in relative coordinates. In example /tppos ~0 ~1 ~0 will teleport you 1 block up
```

**Comment by CloudeLecaw** [comment]
i agree with mrfloris a tpw normaly is not needed because all current TP commands allready support the worldargument.

The command you wanted allready exists its name is just not tpw - create an alias for this if needed:

**You searched for:**
world (normal/nether/end/1/2/3...) (playerName) (-s)
```
Teleports to different world

Requires cmi.command.world.[worldName] in addition to base permission node
```
So to send an other player to your world simply type: /world MyCoolWorld X_MyVictim_X -s
then X_MyVictim_X will be teleported to MyCoolWorld - or do it without the -s then X_MyVictim_X get informed that you send him to MyCoolWorld. Where he is landing? Right at the point that you have defined as spawn for this world. If you wanne send him to a strict position then look in the other commands that can teleport to an other world.
.
.
**_CMI has many other TP commands- take a look here_**
.
.

**These commands can Teleport players to other worlds if someone is in this world allready:**

tp [playerName] (playerName)
```
Teleports to player’s location
```
tpa [playerName] (playerName) (-c)
```
Ask the player if you can teleport to them
```
tpaall
```
Ask all online players to teleport to your location
```
tpahere [playerName] (playerName) (-c)
```
Asks player to accept teleportation to your location

-c will teleport to location when offer was made and not where player is currently.
This requires cmi.teleport.currentlocation permission node from command sender
```
tpall (playerName)
```
Teleports all online players to location
```
tphere [playerName] (playerName)
```
Teleports player to your location
```
tpo [playerName] (playerName)
```
Teleports to player’s location by force
```
tpohere [playerName] (playerName)
```
Teleports player to your location by force
```


**There commands can have a worldvar allready**
tpallworld [worldName] (worldName;x;y;z(;yaw;pitch)) (-a)
```
Teleports ALL players from specific world. Target location can be defined in case
you want to perform this command from console.

Example: /cmi tpallworld LT_Craft_nether LT_Craft;0;150;0

-a will teleport everyone who is located in no longer existing world.
This can happen if you have removed custom world with plugin like
multiverse core and offline player is still located in that world
```
tppos (-p:playerName) [x] [y] [z] (world) (pitch) (yaw) (-rng:[range])
```
Teleports to location

-p: will define player you want to teleport. This will require you to
have cmi.command.tppos.others permission node

-rng: will teleport at random location from center point by given radius

x, y and z can be in relative coordinates. In example
/tppos ~0 ~1 ~0 will teleport you 1 block up
```
tpopos (-p:playerName) [x] [y] [z] (world) (pitch) (yaw) (-rng:[range])
```
Teleports to location by force and will override any area protections
player might have

-p: will define player you want to teleport. This will require you to
 have cmi.command.tpopos.others permission node

-rng: will teleport at random location from center point by
given radius

x, y and z can be in relative coordinates. In example
/tpopos ~0 ~1 ~0 will teleport you 1 block up
```
